### PR TITLE
 Corrected expectedX

### PR DESCRIPTION
**Pull request by cathiechen**
The default scroll position is on the right padding side.

### CODE REVIEW

**Comment by cathiechen** [comment]
cc @fred-wang

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html [hunk 1]
```diff
@@ -68,8 +68,20 @@
 // As browsers differ in the meaning of scrollLeft when
 // in a right-to-left mode, we adjust the expectation
 // to match the semantics of scrollLeft.
-if(scroller.scrollLeft === 0)
-  expectedX = [ -box_width, -(((3*box_width - scroller_width)/2)+ (scrollbar_width/2)), -(((2*box_width)-scroller_width)+scrollbar_width)];
+// As browsers differ in the meaning of scrollLeft when
+// in a right-to-left mode, we adjust the expectation
```
Comment (fred-wang) [inline css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: I think the x-coordinate is nonpositive when box has leftward overflow direction:
https://drafts.csswg.org/cssom-view/#scroll-an-element
and increases rightwards https://drafts.csswg.org/cssom-view/#scrolling-area-origin
so the initial value is 0 per the spec.

So what is calcutated here should actually be the initial value of expectedX and you should test for scroller.scrollLeft  > 0 here to fix the behavior for browsers that don't follow the spec.

nit: please add a space after "if"

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html [hunk 2]
```diff
@@ -68,8 +68,20 @@
...
+// to match the semantics of scrollLeft.
+if(scroller.scrollLeft === 0) {
+  // If scroll bar is on the left side, scroller.scrollLeft won't bigger then 0,
+  scroller.scrollLeft = 1000;
```
Comment (fred-wang) [inline css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: Would just using scroller.scrollLeft = scrollbar_width work here?

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html [hunk 3]
```diff
@@ -68,8 +68,20 @@
 // As browsers differ in the meaning of scrollLeft when
 // in a right-to-left mode, we adjust the expectation
 // to match the semantics of scrollLeft.
-if(scroller.scrollLeft === 0)
-  expectedX = [ -box_width, -(((3*box_width - scroller_width)/2)+ (scrollbar_width/2)), -(((2*box_width)-scroller_width)+scrollbar_width)];
+// As browsers differ in the meaning of scrollLeft when
+// in a right-to-left mode, we adjust the expectation
+// to match the semantics of scrollLeft.
+if(scroller.scrollLeft === 0) {
+  // If scroll bar is on the left side, scroller.scrollLeft won't bigger then 0,
```
Comment (fred-wang) [inline css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: "If vertical scrollbar is on the right side, one can scrollLeft to positive values in order to show the content hidden by the bar?"

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html [hunk 4]
```diff
@@ -68,8 +68,17 @@
 // As browsers differ in the meaning of scrollLeft when
 // in a right-to-left mode, we adjust the expectation
 // to match the semantics of scrollLeft.
-if(scroller.scrollLeft === 0)
-  expectedX = [ -box_width, -(((3*box_width - scroller_width)/2)+ (scrollbar_width/2)), -(((2*box_width)-scroller_width)+scrollbar_width)];
+if(scroller.scrollLeft === 0) {
+  // If scroll bar is on the left side, scroller.scrollLeft won't bigger then 0,
+  scroller.scrollLeft = 1000;
+  if (scroller.scrollLeft == 0) {
+    // Scroll bar is on the left side.
```
Comment (fred-wang) [inline css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: I guess you don't need the comment here if it's explained above

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html [hunk 5]
```diff
@@ -68,8 +68,17 @@
...
+    expectedX = [ -box_width, -(((3*box_width - scroller_width)/2) + (scrollbar_width/2)), -(((2*box_width)-scroller_width) + scrollbar_width) ];
+  } else {
+    // Scroll bar is on the right side.
```
Comment (fred-wang) [inline css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: ditto.

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html [hunk 6]
```diff
@@ -68,8 +68,17 @@
 // As browsers differ in the meaning of scrollLeft when
```
Comment (fred-wang) [inline css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: I think the x-coordinate is nonpositive when box has leftward overflow direction:
https://drafts.csswg.org/cssom-view/#scroll-an-element
and increases rightwards https://drafts.csswg.org/cssom-view/#scrolling-area-origin
so the initial value is 0 per the spec.

So what is calcutated here should actually be the initial value of expectedX and you should test for scroller.scrollLeft  > 0 here to fix the behavior for browsers that don't follow the spec.

**Review by fred-wang — COMMENTED** [review]
LGTM

Can you please add a comment at the beginning of the script:

    // In vertical-rl mode, X corresponds to the block axis and is oriented
    // leftward. Y corresponds to the inline axis and is oriented downward.
    // This assumes that the horizontal scrollbar is on the bottom side.

and convert these arrays to objects in order to make this more readable:

    var expectedX = {
      blockStart: ...,
      blockCenter: ...,
      blockEnd: ...,
    };
    var expectedY = {
      inlineStart: ...,
      inlineCenter: ...,
      inlineEnd: ...,
    };

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html (hunk 2 above)
Comment (cathiechen) [reply css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: done

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html (hunk 4 above)
Comment (cathiechen) [reply css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: done

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html (hunk 5 above)
Comment (cathiechen) [reply css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: done

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html (hunk 3 above)
Comment (cathiechen) [reply css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: Yes.

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html [hunk 7]
```diff
@@ -62,14 +65,40 @@
 var box_width = target.offsetWidth;
 var box_height = target.offsetHeight;
 
-var expectedX = [ ((2*box_width)-scroller_width)+scrollbar_width, ((3*box_width - scroller_width)/2)+ (scrollbar_width/2), box_width ];
-var expectedY = [ box_height, ((3*box_height - scroller_height)/2) + (scrollbar_width/2), ((2*box_height)-scroller_height) + scrollbar_width ];
+var expectedX;
+// If scroll bar is on the left side, scroller.scrollLeft won't bigger than 0,
```
Comment (fred-wang) [inline css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: won't get*

File: css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html [hunk 8]
```diff
@@ -62,14 +65,40 @@
...
+
+var expectedY = {
+  inlineStart: box_height,
+  inlineCenter: ((3*box_height - scroller_height)/2) + (scrollbar_width/2),
+  inlineEnd: ((2*box_height) - scroller_height) + scrollbar_width,
+};
 
 // As browsers differ in the meaning of scrollLeft when
 // in a right-to-left mode, we adjust the expectation
 // to match the semantics of scrollLeft.
```
Comment (fred-wang) [inline css/cssom-view/scrollIntoView-vertical-rl-writing-mode.html]: Can you please update the comment:

"In vertical-rl mode, the scroll x coordinate should be nonpositive per the the spec. Some browsers make it nonnegative, so we adjust the expectation.

**Review by fred-wang — APPROVED** [review]

**Pull request merged**